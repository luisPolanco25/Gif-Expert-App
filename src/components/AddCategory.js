import React, {useState} from 'react';
import PropTypes from 'prop-types'

export const AddCategory = ({setCategories}) => {
    
    
    const [inputValue, setInputValue] = useState('');

    // handleInputChange
    const handleInputChange = (e) => {
        setInputValue(e.target.value);
    }

    // handleSubmit
    const handleSubmit = (e) => {
        e.preventDefault();

        if (inputValue.trim().length > 2) {
        setCategories(categories => [inputValue, ...categories]);
        setInputValue('');
        }
    }  

    return (
        <form onSubmit={handleSubmit}>
            <input 
                type="text"
                value={inputValue}
                onChange={handleInputChange} />
        </form>  
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired,
}