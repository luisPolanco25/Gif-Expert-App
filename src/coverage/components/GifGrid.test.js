import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { GifGrid } from "../../components/GifGrid";
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');


describe('Tests in <GifGrid />', () => {
    
    const category = 'Hello world'
    
    test('should match with its snapshot', () => {
        
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true,
        });
        
        const wrapper = shallow(<GifGrid category={category}/>);
        expect(wrapper).toMatchSnapshot();
        
    });
    
    test('should show items when images are loaded with useFetchGifs', () => {
       
        const gifs = [{
            id: 'ABC',
            url: 'https://localhost/something.jpg',
            title: 'Something',
        },
        {   id: '123',
            url: 'https://localhost/something.jpg',
            title: 'Something',
        }];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false,
        });
        
        const wrapper = shallow(<GifGrid category={category}/>);
        
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
    });

})
