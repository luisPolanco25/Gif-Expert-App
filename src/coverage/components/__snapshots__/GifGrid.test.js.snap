// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <GifGrid /> should match with its snapshot 1`] = `
<Fragment>
  <h3
    className="animate__animated animate_fadeIn"
  >
    Hello world
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Tests in <GifGrid /> should show items when images are loaded with useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate_fadeIn"
  >
    Hello world
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Something"
      url="https://localhost/something.jpg"
    />
    <GifGridItem
      id="123"
      key="123"
      title="Something"
      url="https://localhost/something.jpg"
    />
  </div>
</Fragment>
`;
